src/engine/effects/engineeffect.h

```diff
@@ -26,6 +26,7 @@ class EngineEffect final : public EffectsRequestHandler {
             const QSet<ChannelHandleAndGroup>& registeredInputChannels,
             const QSet<ChannelHandleAndGroup>& registeredOutputChannels);
     /// Called in main thread by EffectSlot
+    // Doesn't deal with ownership; only for conditional debug output
```

Comment:
Perhaps not for that PR, but I'm wondering if this could be done with a build option instead. 
Since `kEffectDebugOutput` needs to be flipped manually when one wants to debug effects, it could as well be a build option.

Reply:
Usually a constant is preffered because unlike with a preprocessor definition, the code that will get optimized out is still built and checked for correctness. That avoids the issue where code silently breaks because it has not been built under CI. Ideally we could instead compile-out the debug destructor by use of prospective destructors. I didn't want to go into that complexity yet though.